scripts/loader.js

```diff
@@ -0,0 +1,348 @@
+"use strict";
+
+const { Client } = require("pg");
+const fs = require("fs");
+const { parse } = require("csv-parse");
+const { finished } = require("stream/promises");
+
+const client = new Client();
+
+let course_sections = [];
+
+const DEBUG = false;
+let query_count = 0;
+
+const loader = DEBUG
+  ? async (query, values) => {
+      let res = await client.query(query, values);
+      if (DEBUG) console.log(`Query ${query_count++} Completed`);
+      return res;
+    }
+  : async (query, values) => client.query(query, values);
+
+const processFile = async (filepath) => {
+  const records = [];
+  const parser = fs.createReadStream(filepath).pipe(
+    parse({
+      // CSV options if any
+    })
+  );
+  parser.on("readable", function () {
+    let record;
+    while ((record = parser.read()) !== null) {
+      // Work with each record
+      records.push(record);
+    }
+  });
+  await finished(parser);
+  return records;
+};
+
+const weekday_map = {
+  M: "Monday",
+  T: "Tuesday",
+  W: "Wednesday",
+  R: "Thursday",
+  F: "Friday",
+  S: "Saturday",
+  U: "Sunday",
+};
+
+const course_type_map = {
+  INT: "Internship",
+  INS: "Independent Study",
+  CMP: "Competition",
+  LEC: "Lecture",
+  LAB: "Laboratory",
+  PRAC: "Practicum",
+  CLD: "Clinic",
+  REC: "Recitation",
+  EXAM: "Examination",
+  SEM: "Seminar",
+  RES: "Research",
+  PRL: "Private Lesson",
+};
+
+const to_db_time = time => `${time}:00`;
+
+const update_course_sections = (
+  dbobj,
+  section_id,
+  section_number,
+  weekday,
+  start_time,
+  end_time,
+  place,
+  course_type
```

Comment:
Functions that take lots of parameters like this happen quite a bit over the file. In TypeScript, you'd create an interface for an object that contains all these things, but JS doesn't have things like that, so I'm not sure how to clean up the number of parameters. All that to say I don't have a recommendation for this, it's probably fine 😅